aten/src/ATen/native/SobolEngineOps.cpp

```diff
@@ -150,7 +150,12 @@ Tensor& _sobol_engine_initialize_state_(Tensor& sobolstate, int64_t dimension) {
     }
   }
 
-  Tensor pow2s = at::pow(2, at::native::arange((MAXBIT - 1), -1, -1, sobolstate.options()));
+  Tensor pow2s = at::pow(2, at::native::arange(
+    (MAXBIT - 1), -1, -1, 
+    typeMetaToScalarType(sobolstate.options().dtype()), 
+                         sobolstate.options().layout(), 
+                         sobolstate.options().device(), 
+                         sobolstate.options().pinned_memory()));
```

Comment:
This is always `false`, per the definitions of `Tensor::options()`

```
inline TensorOptions Tensor::options() const {
  return TensorOptions().dtype(dtype())
                        .device(device())
                        .layout(layout());
}

```

Reply:
You mean `pinned_memory`? why?

Reply:
I mean, we cant know if was set up to any value or not so we are just following the existing logic and passing it down.

Reply:
I mean `pinned_memory`. We know that it is not set because `options` never sets it. Your code is written defensively but it is confusing because it implies that `pinned_memory` could be set, when it is never set.

As I've mentioned earlier, the contract for `_like` functions is they preserve dtype, device and layout, but not pinnedness or requires grad-ness. I'm not sure there's a good a priori justification for this, but post facto justification is something like, you want to explicitly say when you want pinned memory, you shouldn't reallocate outputs into pinned memory just because inputs were pinned.